contracts/IDelegator.sol

```diff
@@ -2,18 +2,41 @@
 pragma solidity ^0.8.0;
 
 interface IDelegator {
-    function upgradeDelegate(address newDelegateAddress) external;
+    function updateAddress(
+        address newDelegateAddress,
+        address newResultAddres,
+        address parametersAddress
+    ) external;
 
-    function getResult(uint8 id) external view returns (uint32);
+    function setIDName(string calldata name, uint8 _id) external;
 
-    function getJob(uint8 id)
+    function getResult(bytes32 _name) external view returns (uint32);
+
+    function getJob(bytes32 _name)
```

Comment:
Why is this here? Does any contract need it?

Reply:
Nope, Will need to remove it